### DISCUSSION TITLE
2024-04-26 - Client Love 💋 - Chewing the Fat 📈

### DISCUSSION THREAD

**Discussion by bradystroud** [SSWConsulting/SSW.Rules.Content#8432] [Chewing the Fat]
Hey SSWers,

Let's talk about showing client love...

https://ssw.com.au/rules/weekly-client-love/

How do you show your clients love?
Do you naturally keep in touch with old clients?

**Comment by calumjs** [comment]
it would be cool if AI could manage this process

we have a lot of details on every client e.g. project skills, communications, backlogs, repos, documents, etc. plus general details about their company

we could then get latest tech and industry news, and run it all through gpt to create a few suggested conversation starters for each dev relevant to their past projects automatically sent to them each week. 

then we can pick out the best idea, use it as a draft, and make the call

**Comment by duanxinhuan** [comment]
Glad to help but I think reaching out old client is not the best way.
My neibour Kevin he is a sales at Google Austalian, even him is selling product to Goverment.

If we look at the figure:
<img width="972" alt="image" src="https://github.com/SSWConsulting/SSW.Rules.Content/assets/48227836/5f2c6285-4002-4be4-89f3-0eae1cd527b0">
Aussie goverment spending to GDP is almost 25% percent, reaching out to public sector would ve the most efficient way to increase utilitization rate. Furthermore aus goverment is also planning to spend billions of dollars on IT:
https://www.bdo.com.au/en-au/insights/technology-media-entertainment-telecommunications/outlook-on-australia-s-tech-sector-bdo-s-international-tech-hubs-2023

Also it's a tough time for getting client from private sector at the moment because interest rate is too high:
If I can just get 5% from back why would I invest money in a software? Why would I loan from bank to build a software?

We need to be good at dealing with public sector and get big fish for us(Also keep in mind our biggest source of client has been public sector in the last few years, before that it was financial sector).

**Comment by PothieuG** [comment]
Thanks for the rules !

For someone who is not familiar with this kind of process, it is always interesting to see and understand how to maintain contact and perhaps create new opportunities with former clients!

**Comment by Freego1783** [comment]
Not really used to doing this and it's not something I think I'll like to do but I do understand why it can be beneficial. A little worried about people making mistakes and contacting the same client though.

**Comment by tiagov8** [comment]
The effort to get new clients if definitely bigger than to keep existing. 
We could encourage developers to involve **designers** in current/past projects as a way to improve (even if slightly) their UI/UX (when applicable)

**Comment by leahy268** [comment]
I find just reaching out every now and then to an old client to ask them things like how the work you did for them in the past is going a great way to get new work.  It's a simple remember me kind of thing.

Also would be great to get an integration with TimePro and Ingram, so that we can really start selling licenses/Azure to clients.  I would love to start pushing that stuff which is just an extra way that a client needs to contact us.

We also now qualify to apply for at least 3 Microsoft Azure Specialization's that I know of.  All of these push us further up the list of contacts for different products.  It does require some effort on our part, but while utilization is low would be a good time.

**Comment by miichaelsmedley** [comment]
In my view, sales is as challenging as the most intricate technical problem. The best salespeople often command salaries comparable to, if not exceeding, those of top developers. Unlike many coding roles which can be solitary, sales is fundamentally about engagement and interaction. It's a field where even seasoned professionals face the possibility of rejection in every conversation with potential clients.

Success in sales is driven by numbers and continuous improvement based on feedback. The public is an unforgiving judge, quick to indicate whether you, your product, or your service falls short. Handling rejection is a part of the job that never gets easier. Over the coming period, I'll be reaching out to hundreds of new, existing, and former clients and contacts. Watch this space.

Interestingly, I’ve had salespeople who struggled with this aspect of the job. It’s a tough challenge, and I’m curious to see how developers might handle the same tasks.

Stay tuned!

**Comment by joshbermanssw** [comment]
I havent yet been on any client work since arriving at SSW, however i do think the practice of creating connections with clients is essential for maintaining business in the future. Work becomes much more enjoyable when you look forward to working with someone. 

I dont necessarily think the client reach out task should occur on a weekly basis as this can become almost spammy, perhaps rather a bi-monthly task at shortest.

**Comment by jakebayliss** [comment]
I think this is great if we are reaching out with a good, thoughtful idea or just checking in on a past project/client. In terms of trying to win opportunites I think it should be sales people reaching out as they are a lot more practiced. My last job had 4-5 really good sales people and they often didnt want devs speaking too much as they (we 🙃) would say the wrong things sometimes. However I see no downside in reaching out to see how people are going - Something I should do more often!

**Comment by KostaMadorsky** [comment]
(checked by Adam)

Devs keeping in touch with past client is a good thing. It should be natural and organic, then no devs would be saying the same thing.

The timing is important. Imagine 3 devs contacting the client at the same time, would not be good. So it is good that the rule mentions to look up the Power BI report.

Warning: Account managers should coordinate important messages. If a client is asked multiple times by different developers the same thing, it would have the opposite effect.

**Comment by camillars** [comment]
Building connections can be challenging for introverts, but it's the right sales approach as it has more benefits than negatives. 🙂 Since multiple people can reach out to the same client, it's important to follow the SugarLearning step to check the Power BI report first.

This actually made me add a previous Google Ads client on LinkedIn to check in how his campaigns are going and if his website is performing well (I suggested a website revamp to improve conversions at the time) 🙂

**Comment by Calinator444** [comment]
It's a good strategy but I would hope that not every interaction with a client is strictly professional or centred around securing more work. I think our job as consultants is to provide clients with the best solutions to meet their that falls within the scope of their budget and resources. I'm assuming you would build trust organically with clients and thus receive more work if you're able to deliver on this. 

Granted I haven't been on any client projects yet, this may be more difficult to do than it seems.

**Comment by TobyChurches** [comment]
Its a good rule but like many people have mentioned, we need a way to make sure that we aren't contacting the same people multiple times since this would have the adverse effect. Maybe we could have a place were we could log this after contacting in the CRM?

**Comment by danielmackay** [comment]
Keeping a good relationship with your client is paramount.  I always try to build a relationship with key personal that involves other things besides work.

Historically, I thought keeping in touch with the client was more of an account manager or sales job.  To some extent, it is but after more thought, developers can offer more grounded and practical advice to solve clients issues.

**Comment by stephanfakossw** [comment]
I think it's super important for account managers and BD people to reach out to existing clients to create repeat business. BD & sales are not an easy skill, often taking years and plenty of experience to master.
I'm not convinced it should be developers reaching out to clients - they typically lack sales skills even if they can talk. Especially developers at the beginning of their career... Bad reach outs can shut the doors rather than open them.

**Comment by bahjatalaadelssw** [comment]
Video and marketing is still the most effective way to attract work. We don't engage with the content we publish. A more agile process would give us more visibility.

**Comment by awaltos** [comment]
From a dev perspective, I think reaching out to clients in a casual way is a good idea for the purpose of maintaining a good relationship - as @leahy268 mentioned, a simple check-in of "How is that thing going that we did for you?" is a great way to do this. Devs are generally more trusted by clients, than for example salespeople, especially if that dev has met client expectations on some work in the past. Building on this level of trust with semi-regular check-ins is gold standard for fostering relationships.

There is the potential to cause friction if the client is getting a lot of 'check-ins' from people within the same company, so working together and having visibility on conversations is important to mitigate that risk. I also think that Account managers still play an important role managing those relationships in a sales-driven way as they not only have the skills but have the context about when to contact them and how often to do so.

**Comment by ricksu978** [comment]
Agree that maintaining relationship is important, but this should be handled with care with right people with right skill at the right frequency.

**Comment by chrisschultzssw** [comment]
As others have said, too much of this could cause issues like multiple people contacting the same client, or poor-quality comms going out to our clients.

There are simpler ways to stay engaged with clients, e.g. connecting on LinkedIn and interacting with their posts.

**Comment by tombui99** [comment]
Contacting old clients is a nice gesture, but it can backfire if it becomes too pushy. Over the years, I've built solid relationships with past clients by being personable and keeping conversations lively. Not too long ago, I bumped into one of my former clients at a mall, and we had a laid-back, lengthy chat. However, landing new works can be tough because most of the contacts I've worked with before, especially as a developer, is not in the position to allocate more work due to cost concerns.

**Comment by jeoffreyfischer** [comment]
It's a good reminder that we should keep in touch with former clients - in my case it's just obvious but I understand that not everyone behaves the same.

**Comment by amankumarrr** [comment]
Maintaining a good relationship with clients is crucial. Sending automated birthday emails with the latest offerings is a great way to keep them engaged and increase our visibility.

**Comment by AlexRothwell** [comment]
Personally I've only really formed a relationship with 1 of my previous clients but have not kept in touch recently. This has been a good reminder for me to see how they've been finding the work we did. Especially in that case, there was some technical choices I wouldn't have made myself so I'm keen to see how it's turned out

**Comment by Aibono1225** [comment]
Showing client love and keeping contact are important. However reaching out weekly could be too much, as not everyone like this frequency

**Comment by AnkyChe** [comment]
I love this topic! Developers possess unique knowledge that clients often require. By demonstrating affection for the client and engaging in discussions about technical topics, developers can leave a lasting impression.

**Comment by jimmidier** [comment]
Showing client love and keeping relationship warm is important to seek potential opportunities

**Comment by KristenHu** [comment]
Great topic! Sales can be really hard, it's great to find new ways to improve utilization, however weekly greeting may be too much, it may Leave a negative impression

**Comment by tino-liu** [comment]
It's very nice to reconnect with colleagues from the past (And even friends from university). Catching up and chatting with them is not only enjoyable but also enlightening... I learned quite a few . Plus, it helps ensure they remember me as well. I believe the experience would be similar with clients.

**Comment by jerryluossw** [comment]
Maintain good relationships with regular clients are important, and I think it's better be under the guidance of sales or account manager, at the same time, for same clinet should not be weekly.

In my opinion, the best way for devs to sell is:

1. During the project development process, chat to clients or dig deeper into their requirements to find new opportunities.
2. For regular clients, introduce the company's new products or latest solution(Like AI) to them periodically.

**Comment by Dhruv-0987** [comment]
Maintaining good professional relationships with everyone is important especially with clients. 

Clients generally enjoy talking to developer more then sales people because we are not trying to sell them anything 😄.

**Comment by drwharris** [comment]
What do we do when we dont win work? Do we follow up ? Do we make changes based on feedback?

**Comment by AntPolkanov** [comment]
Staying tight with clients is key. It's all about keeping the vibe positive and making sure they feel heard, which sets the stage for more collaborations down the road.

**Comment by YalalCakirSSW** [comment]
This rule makes me realize that there are a lot of people with whom I haven't talked for a long time. 😅 Keeping in touch is great.
I don't have any client experience to do this thing or comment about it, but I would like to change it.

**Comment by babakamyljanovssw** [comment]
For me I haven't done any client work yet, but having professional relationship with clients is essential skill, and I would love to have one soon 😊

**Comment by christoment** [comment]
I think everyone has their own skill in where they excel. Developers might not be the right person to do reach out - Sales or account managers is the better person to do this since they have more skills in sales and business than most developers.

Developers can cover more ground in technical concerns in the conversation, but I think people with excellent sales skills should take the lead on this so we don't lose any good opportunities 😄 

For developers, this make sense:

```c#
while (!client.IsHappy()) {
    BuyCoffee(client);
}
```

but alas, that won't work in real life 😄

**Comment by steven0x51** [comment]
Maintaining a good relationship with clients is crucial. Reaching out doesn't mean spamming their inbox, rather, it involves genuine gestures such as natural greetings and discussing common interests. Moreover, this outreach can extend beyond just the client to colleagues and friends.

**Comment by tomek-i** [comment]
It's important to maintain good relations. Our goal is to engage naturally, avoiding any forced or frequent outreach that feels artificial.

When working with clients, casual conversations can be incredibly insightful. They provide insights we can use to spark future conversations or discussions organically.

**Comment by GordonBeeming** [comment]
Reaching out to old customers (and old friends even) is something that most people don't do enough. It's just part of general relationship building and networking.
I don't think every interaction needs a follow-up email, although sometimes there is opportunities that come out of the chat, and I think it's perfectly fine for those to make their way into an email 🙂

**Comment by cvionnet** [comment]
Staying in touch is an important yet often overlooked task.
It helps maintain connections and provides updates on project progress while keeping relationships alive.
This practice can be extended beyond just clients, it can include past colleagues, university friends, etc

**Comment by jimmy1992abc** [comment]
Good business meets the client's demand and the top business design customer's demand.
Everyone wants to be the top business, but it can only be built based on a good business.

**Comment by JeanThirion** [comment]
All the things in the Rule are cool, but here a few others:
- Go for a coffee/beer after hours if you can (1x year for good clients)
- Make sure they're happy even when you're no longer on the project (hey, how's Jean doing ?)
- Let them know you're keeping an eye out ("I know Charles and Jeoffrey are working for you now, don't worry I'm keeping an eye on comms :))
- Reach out after X months in Prod to check "Hey, how's the product going ? Everybody happy ?"
- Offer audits/reviews - "Hey Angular 1324 is out, you probably want to updated because XYZ"

But really the best thing is to genuinely care for the people... Take the time to understand what they really want to achieve past the project, understand how you can truly help.
Clients will go back to **you** if they can trust you to deliver.

**Comment by samsmithssw** [comment]
I think keeping casual conversations with clients as a dev, providing interesting insights and giving them value will make them appreciate you more and feel a sense of reciprocity back. It means that you'll be in the forefront of their minds when they need help with tech-related problems - which is only a good thing for us. 

Repeat business is the best source of work - they already know us, they know how good we are, they know how we operate.

**Comment by tkapa** [comment]
Good casual conversation is a great way to do client love. I like to chat and make easy jokes.

I tend not to keep in touch with a client after the fact unless I liked that client as a person. To me, even the idea that I would reach out feels kind of salesy, in a way.

**Comment by vladislav-kir** [comment]
It is important to keep connection with the previous clients going. I think it is a great idea for developers to do reaching out. However, I do not like the idea of having it a task that every developer has to regularly perform and using GPT for it, since, in my opinion, the interactions might feel unnatural and non genuine and can lead to the opposite of the desired effect.

**Comment by ozair-ashfaque** [comment]
Good communication is crucial for a strong relationship. As a developer, your technical skills can impress and make clients happy. This might make them want to work with you again in the future. But Sales or account managers are usually better at talking to clients and reaching out to them because they are good at communicating. However, developers can still join Account managers in informal meetings with clients.

**Comment by MarcusIrmscherSSW** [comment]
I think reaching out to clients is a useful thing but needs to be done sparingly. If done too often it can just be an annoyance.

**Comment by bettybondoc** [comment]
Fostering good relationships with clients is important, but the current approach specified in the SugarLearning item might need adjustment. For instance, if a developer has only ever worked with two clients, frequent calls to the same individuals could become overly repetitive and potentially unwelcome. Similarly, a client who has interacted with three developers might find it irritating to receive calls from multiple people. If only one developer makes the call, the task for the other two becomes redundant and cumbersome. Additionally, requiring everyone else to email three people weekly seems excessive. We should consider revising the frequency or method of these interactions to maintain the balance between keeping in touch and respecting everyone's time.

**Comment by PennyWalker** [comment]
I think it's great to reach out to old clients, especially if we have something cool to offer them like an interesting bug fix, new development idea, or free Uperpowers tickets. It's important for the love to be meaningful and not an 'ask' otherwise it's going to bug the client that they're hearing form you instead of exciting them. Done well, it can lead to work - I know Mehmet has been really successful with this in the past, and all devs should be trying to build relationships, not just relying on Account Managers - it's the devs they trust. :)

**Comment by uly1** [comment]
The key is just to show interest, and provide as much value as possible. Give them links to cool vids or articles, or ask questions about if they're planning to update xyz... Adding them on Teams is also valuable

**Comment by MattParkerDev** [comment]
Keeping in touch with our clients is important.
